Bottom: 011d59af51166bf15670275f31d60ff82a5c9ec6
Top:    f482ccb80350f54e526eee4c9595c6b489b0052e
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-11 15:14:09 +0000

Win32 code fixes (TO BE SPLIT INTO SEPARATE PATCHES)


---

diff --git a/src/borast/borast-bentley-ottmann.c b/src/borast/borast-bentley-ottmann.c
index c64a120..4f2937f 100644
--- a/src/borast/borast-bentley-ottmann.c
+++ b/src/borast/borast-bentley-ottmann.c
@@ -48,12 +48,18 @@
 #include <glib.h>
 
 #include "polygon.h"
-#include <GL/gl.h>
 #include "hid_draw.h"
+
+#ifdef WIN32
+#   define WIN32_LEAN_AND_MEAN 1
+#endif
+
+#include <GL/gl.h>
 #include "hid/common/hidgl.h"
 
 #include "sweep.h"
 
+
 #define _borast_error(x) (x)
 
 #define DEBUG_PRINT_STATE 0
diff --git a/src/compat.c b/src/compat.c
index 6092764..dd1910b 100644
--- a/src/compat.c
+++ b/src/compat.c
@@ -57,6 +57,7 @@ random (void)
 #endif
 
 #if !defined(HAVE_DLFCN_H) && defined(WIN32)
+#define WIN32_LEAN_AND_MEAN 1
 #include <windows.h>
 
 void *
diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index e6b1eaf..26cb769 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -35,6 +35,7 @@
 #ifdef WIN32
 #   define WIN32_LEAN_AND_MEAN 1
 #   include <windows.h>
+#   define _GLUfuncptr void *
 #else
     /* The Linux OpenGL ABI 1.0 spec requires that we define
      * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
diff --git a/src/hid/common/hidgl_package_acy_resistor.c b/src/hid/common/hidgl_package_acy_resistor.c
index 97434e4..5e5aedb 100644
--- a/src/hid/common/hidgl_package_acy_resistor.c
+++ b/src/hid/common/hidgl_package_acy_resistor.c
@@ -12,7 +12,9 @@
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include "data.h"
 
-#ifndef WIN32
+#ifdef WIN32
+#   define WIN32_LEAN_AND_MEAN 1
+#else
 /* The Linux OpenGL ABI 1.0 spec requires that we define
  * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
  * in order to get prototypes:
diff --git a/src/hid/common/hidinit.c b/src/hid/common/hidinit.c
index 9f05800..ae8b3b8 100644
--- a/src/hid/common/hidinit.c
+++ b/src/hid/common/hidinit.c
@@ -16,6 +16,7 @@
 #include <unistd.h>
 
 #if defined(WIN32) && defined(HAVE_WINDOWS_H)
+#define WIN32_LEAN_AND_MEAN 1
 #include <windows.h>
 #endif
 
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 30202e4..bd9c2e7 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -18,7 +18,9 @@
 #include "quad.h"
 #include "object3d.h"
 
-#ifndef WIN32
+#ifdef WIN32
+#   define WIN32_LEAN_AND_MEAN 1
+#else
 /* The Linux OpenGL ABI 1.0 spec requires that we define
  * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
  * in order to get prototypes:
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index c68b9d3..8f47a3a 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2092,6 +2092,7 @@ REGISTER_FLAGS (ghid_main_flag_list)
  * footprint libraries.
  */
 #ifdef WIN32
+#define WIN32_LEAN_AND_MEAN 1
 #include <windows.h>
 #include <winreg.h>
 #endif
diff --git a/src/hid/gtk/gui-library-window.c b/src/hid/gtk/gui-library-window.c
index 2901c6c..3c6d7ff 100644
--- a/src/hid/gtk/gui-library-window.c
+++ b/src/hid/gtk/gui-library-window.c
@@ -67,6 +67,8 @@
 #include <dmalloc.h>
 #endif
 
+#include <libgen.h> /* For basename() */
+
 static GtkWidget *library_window;
 
 #include "gui-pinout-preview.h"
@@ -621,7 +623,9 @@ create_lib_tree_model (GhidLibraryWindow * library_window)
 	tok_start = tok_end;
 	if (*tok_start == '/')
 	  tok_start++;
-	tok_end = strchrnul(tok_start, '/');
+	tok_end = strchr(tok_start, '/');
+        if (tok_end == NULL)
+          tok_end = tok_start + strlen (tok_start);
       }
     while (*tok_start != '\0');
 
diff --git a/src/hid/gtk/object3d.c b/src/hid/gtk/object3d.c
index ddcf35e..fade34d 100644
--- a/src/hid/gtk/object3d.c
+++ b/src/hid/gtk/object3d.c
@@ -13,7 +13,9 @@
 
 
 
-#ifndef WIN32
+#ifdef WIN32
+#   define WIN32_LEAN_AND_MEAN 1
+#else
 /* The Linux OpenGL ABI 1.0 spec requires that we define
  * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
  * in order to get prototypes:
@@ -244,6 +246,7 @@ object3d_export_to_step (object3d *object, char *filename)
   FILE *f;
   time_t currenttime;
   struct tm utc;
+  struct tm *tmp;
   //int next_step_identifier;
 
   f = fopen (filename, "w");
@@ -254,7 +257,9 @@ object3d_export_to_step (object3d *object, char *filename)
     }
 
   currenttime = time (NULL);
-  gmtime_r (&currenttime, &utc);
+  // gmtime_r (&currenttime, &utc);
+  tmp =  gmtime (&currenttime);
+  utc = *tmp;
 
   fprintf (f, "ISO-10303-21;\n");
   fprintf (f, "HEADER;\n");
diff --git a/src/lrealpath.c b/src/lrealpath.c
index 8977b40..29910ea 100644
--- a/src/lrealpath.c
+++ b/src/lrealpath.c
@@ -66,7 +66,7 @@ extern char *canonicalize_file_name (const char *);
 #else
   /* cygwin has realpath, so it won't get here.  */ 
 # if defined (_WIN32)
-#  define WIN32_LEAN_AND_MEAN
+#  define WIN32_LEAN_AND_MEAN 1
 #  include <windows.h> /* for GetFullPathName */
 # endif
 #endif
diff --git a/src/polygon.c b/src/polygon.c
index e0721ea..265593b 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -2042,12 +2042,13 @@ pv_outline_callback (const BoxType * b, void *cl)
 
       if (element_name != NULL && pin_number != NULL)
         {
-          char *tmp;
+          feature_name = NULL; // Win32
+//          char *tmp;
 
-          feature_name = tmp = malloc (strlen (element_name) + 1 + strlen (pin_number) + 1);
-          tmp = stpcpy (tmp, element_name);
-          *(tmp++) = '-';
-          tmp = stpcpy (tmp, pin_number);
+//          feature_name = tmp = malloc (strlen (element_name) + 1 + strlen (pin_number) + 1);
+//          tmp = stpcpy (tmp, element_name);
+//          *(tmp++) = '-';
+//          tmp = stpcpy (tmp, pin_number);
         }
       else
         {
